File: backend/app/models/question.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from ..core.database import Base


class Question(Base):
    __tablename__ = "questions"

    id = Column(Integer, primary_key=True, index=True)
    knowledge_base_id = Column(Integer, ForeignKey("knowledge_bases.id"), nullable=False, index=True)
    knowledge_point_id = Column(Integer, ForeignKey("knowledge_points.id"), nullable=False, index=True)
    question_type = Column(String(50), nullable=False, index=True)
    stem = Column(Text, nullable=False)
    options = Column(Text, nullable=True)
    answer = Column(String(100), nullable=False)
    analysis = Column(Text, nullable=True)
    difficulty = Column(String(20), default="medium", nullable=False, index=True)
    ai_raw_response = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: frontend/src/api/question.ts
import request from './request'

export interface Question {
  id: number
  knowledge_base_id: number
  knowledge_point_id: number
  question_type: string
  stem: string
  options: { key: string; text: string }[]
  answer: string
  reference_answer: string | null
  analysis: string | null
  difficulty: string
  created_at: string
  updated_at: string
}

export function getQuestions(params?: {
  knowledge_base_id?: number
  knowledge_point_id?: number
  question_type?: string
  difficulty?: string
}) {
  const qs = new URLSearchParams()
  if (params?.knowledge_base_id) qs.set('knowledge_base_id', String(params.knowledge_base_id))
  if (params?.knowledge_point_id) qs.set('knowledge_point_id', String(params.knowledge_point_id))
  if (params?.question_type) qs.set('question_type', params.question_type)
  if (params?.difficulty) qs.set('difficulty', params.difficulty)
  const suffix = qs.toString() ? `?${qs.toString()}` : ''
  return request.get(`/api/questions${suffix}`) as Promise<Question[]>
}

export function getQuestion(id: number) {
  return request.get(`/api/questions/${id}`) as Promise<Question>
}

export function deleteQuestion(id: number) {
  return request.delete(`/api/questions/${id}`) as Promise<{ success: boolean; message: string }>
}

export interface GeneratedQuestion {
  id: number
  question_type: string
  stem: string
  options: { key: string; text: string }[]
  answer: string
  reference_answer: string | null
  analysis: string | null
  difficulty: string
}

export function generateQuestions(knowledgePointId: number, questionTypes: string[], count: number) {
  return request.post('/api/ai/generate-questions', {
    knowledge_point_id: knowledgePointId,
    question_types: questionTypes,
    count,
  }) as Promise<{ created_count: number; skipped_count: number; questions: GeneratedQuestion[] }>
}
